crest/Assets/Crest/Crest/Scripts/LodData/RegisterLodDataInput.cs

```diff
@@ -124,6 +121,7 @@ public bool Validate(OceanRenderer ocean)
     /// <summary>
     /// Registers input to a particular LOD data.
     /// </summary>
+    [ExecuteAlways]
```

Comment:
Thanks.. so accessing rend.material makes it create one? Any thoughts what i can do to resolve this?

Reply:
Right now registration requires the user to have the mesh renderer setup correctly before adding the register component. And it is very difficult to recover from there.

We would need an isValid flag just to make things simpler...

The most robust solution would be to handle all this ourselves (we add both mesh and material fields and create mesh renderer). Then we could use OnValidate to register if it fails in OnEnable. But that is a pretty significant change. The extra benefit is we can setup the mesh renderer how we know it should be and users would get updates from bug fixes. Right now we have no control.

Or we always check these values if isValid is false when it is running in the editor before using them and do some healing. This one might be a bit flakey.

Last is an update function which runs only in the editor and does a delayed registration once everything is valid. But we still need to keep check if user changes the material or something and de-reregister and set invalid appropriately.

It seems this is already a problem but it was a safe assumption not to handle it because of enter/exist play mode purging bad state.

Reply:
To add to the "robust solution" point, we could try something like `Graphics.DrawMesh`, `Graphics.DrawMeshNow` or `Graphics.DrawMeshInstanced` to avoid an extra component.

Reply:
I think maybe polling the data off the object each editor update instead of relying on cached data might be the way forward? I had to do this in a few other places as well. You could easily break the gameobject apart and put it back together in arbitrary ways as a user and expect the system to keep up and handle it gracefully.

Your notes about setting up the object from code probably warrants further discussion. it could be an option, but i think the user could still break the system easily if it doesnt constantly poll the data, so i might look at that for now.

Reply:
Alright i think i resolved this - i poll for renderer/material/queue so this stuff stays live, and i tried different combinations of setting up ocean inputs and it seemed to handle all cases that i tested.

It adds a Update() function which i dont love adding, but it does very little work in standalone builds.